### ISSUE TITLE
PGP Signing & $HOME Not Working 

### ISSUE THREAD

**Issue by elproducto** [devopsfaith/docker-fpm#1]
Thanks for creating this process to leverage FPM in Docker.  I want to point out that I am running into issue when attempting to sign RPMs.  I have provide output of error below and a representation of my Makefile.  I have made sure I have the right password, not sure what else I could be missing. Build works if I disable signing of RPM

I also want to point out when is use the $HOME variable to mount volume in Makefile, as  demoed in blog post, build fails. It works when I enter full path instead.   I also had to remove the 1.9.3 from docker file instruction to get FPM installed on image generated. Otherwise would end in error

**GPG Error**
Enter pass phrase: 
gpg: WARNING: unsafe ownership on homedir `/root/.gnupg' {:level=>:info}
gpg: skipped "24C138BA64AE9697": No secret key {:level=>:info}
gpg: signing failed: No secret key {:level=>:info}
Pass phrase check failed or gpg key expired {:level=>:info}
Process failed: rpmbuild failed (exit code 1). Full command was:["rpmbuild", "-bb", "--define", "buildroot /tmp/package-rpm-build-cb93ac1a8d6f01b2225371c92a742f1989409c0e0f0e94355ed3df8c6cce/BUILD", "--define", "_topdir /tmp/package-rpm-build-cb93ac1a8d6f01b2225371c92a742f1989409c0e0f0e94355ed3df8c6cce", "--define", "_sourcedir /tmp/package-rpm-build-cb93ac1a8d6f01b2225371c92a742f1989409c0e0f0e94355ed3df8c6cce", "--define", "_rpmdir /tmp/package-rpm-build-cb93ac1a8d6f01b2225371c92a742f1989409c0e0f0e94355ed3df8c6cce/RPMS", "--define", "_tmppath /tmp", "--sign", "/tmp/package-rpm-build-cb93ac1a8d6f01b2225371c92a742f1989409c0e0f0e94355ed3df8c6cce/SPECS/mypackage.spec"] {:level=>:error}

**Makefile**
`VERSION := 1.1.8.1
PKGNAME := mypackage
LICENSE := Proprietary
VENDOR= mypackage
URL := https://www.test.com
RELEASE := 1
USER := tcm
ARCH := amd64
DESC := My first package
MAINTAINER := test@test.com
DOCKER_WDIR := /tmp/fpm
DOCKER_FPM := fpm
FPM_OPTS=-s dir -v $(VERSION) -n $(PKGNAME) \
  --license "$(LICENSE)" \
  --vendor "$(VENDOR)" \
  --maintainer "$(MAINTAINER)" \
  --architecture $(ARCH) \
  --url "$(URL)" \
  --description  "$(DESC)" \
  --verbose

DEB_OPTS= -t deb --deb-user $(USER)

RPM_OPTS =--rpm-user $(USER) \
  --rpm-sign

run: ## Run container
	docker run --rm -it -v "${PWD}/rpmmacros:/root/.rpmmacros" -v /home/tcm/.gnupg:/root/.gnupg \
		-v "${PWD}:${DOCKER_WDIR}" -w ${DOCKER_WDIR} ${DOCKER_FPM}:rpm -t rpm ${RPM_OPTS} \
		--iteration ${RELEASE}.iACS \
		-C /tmp/fpm \
		${FPM_OPTS}`

**Comment by taik0** [comment]
Hi @elproducto 
You can use the images directly from the [docker hub](https://hub.docker.com/r/devopsfaith/fpm).

This PGP problem I would say that comes from using a `.gnupg` generated with GPG 1.x/2.0 vs GPG 2.1 or the other way around.

**Comment by elproducto** [comment]
There is an issue with the pull from Docker Hub(Output below).  The GPG on the docker host is 2.2.4, the  Centos:7.6.1810 container has GPG 2.0.2. I tested installing GPG 2.2.9 inside the container by adding RUN & one line from the following Gist, and still got the same.  I also tried changing gpg_name in rpmmacros file to the the key ID from username, with the same outcome.

https://gist.github.com/simbo1905/ba3e8af9a45435db6093aea35c6150e8

**Docker Pull Error**
Using default tag: latest
Error response from daemon: manifest for devopsfaith/fpm:latest not found
